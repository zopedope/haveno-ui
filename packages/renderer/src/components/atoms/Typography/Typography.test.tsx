// =============================================================================
//  Copyright 2022 Haveno
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// =============================================================================

import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { BodyText, Heading, InfoText, LabelText } from ".";

describe("atoms::Typography", () => {
  describe("::BodyText", () => {
    it("renders without exploding", () => {
      const { asFragment } = render(<BodyText>body text</BodyText>);
      expect(asFragment()).toMatchSnapshot();
    });
    it("renders heavy variant exploding", () => {
      const { asFragment } = render(<BodyText heavy>body text</BodyText>);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("::Heading", () => {
    it("renders h1 by default", () => {
      const { asFragment } = render(<Heading>heading text</Heading>);
      expect(asFragment()).toMatchSnapshot();
    });
    it("renders h2", () => {
      const { asFragment } = render(<Heading order={2}>heading text</Heading>);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("::InfoText", () => {
    it("renders without exploding", () => {
      const { asFragment } = render(<InfoText>Info text</InfoText>);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("::LabelText", () => {
    it("renders without exploding", () => {
      const { asFragment } = render(<LabelText>Label text</LabelText>);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
