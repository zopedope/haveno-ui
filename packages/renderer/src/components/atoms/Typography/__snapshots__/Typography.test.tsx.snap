// Vitest Snapshot v1

exports[`atoms::Typography > ::BodyText > renders heavy variant exploding 1`] = `
<DocumentFragment>
  <div
    class="mantine-Text-root mantine-1ikb0mj"
  >
    body text
  </div>
</DocumentFragment>
`;

exports[`atoms::Typography > ::BodyText > renders without exploding 1`] = `
<DocumentFragment>
  <div
    class="mantine-Text-root mantine-1ud14e1"
  >
    body text
  </div>
</DocumentFragment>
`;

exports[`atoms::Typography > ::Heading > renders h1 by default 1`] = `
<DocumentFragment>
  <h1
    class="mantine-Title-root mantine-1krktmh"
  >
    heading text
  </h1>
</DocumentFragment>
`;

exports[`atoms::Typography > ::Heading > renders h2 1`] = `
<DocumentFragment>
  <h2
    class="mantine-Title-root mantine-u6vx0o"
  >
    heading text
  </h2>
</DocumentFragment>
`;

exports[`atoms::Typography > ::InfoText > renders without exploding 1`] = `
<DocumentFragment>
  <div
    class="mantine-Text-root mantine-g6pedd"
  >
    Info text
  </div>
</DocumentFragment>
`;

exports[`atoms::Typography > ::LabelText > renders without exploding 1`] = `
<DocumentFragment>
  <label
    class="mantine-Text-root mantine-1djydci"
  >
    Label text
  </label>
</DocumentFragment>
`;
